lib/Doctrine/DBAL/Schema/AbstractAsset.php

```diff
@@ -166,7 +168,7 @@ public function getName()
             return $this->_namespace . '.' . $this->_name;
         }
 
-        return $this->_name;
+        return $this->_name ?? '';
```

Comment:
Would it become much worse if we make the method return a nullable string? Not sure if assets with empty names exist.

Reply:
I'd say that an asset needs to have a name or throw an exception.

Reply:
@morozov If we change getName() to return ?string we immediately get 70 PHPStan errors on other places. This is because basically all code assumes this method returns string (and there is e.g. a lot of string concatenation happening).

Reply:
> I'd say that an asset needs to have a name or throw an exception.

For some types of assets, it's fine to not have a name. E.g. indices are identified by their definition, not by name.

> If we change getName() to return ?string we immediately get 70 PHPStan errors on other places.

Let's leave it as is for now and revisit later :+1: